src/agent/agent.c

```diff
@@ -2057,6 +2061,16 @@ bool agent_stop(Agent *agent) {
 static bool agent_connect(Agent *agent) {
         peer_bus_close(agent->peer_dbus);
 
+        // resolve FQDN again in case system changed
+        // e.g. hirte controller has been migrated
+        if (agent->orch_addr != NULL) {
+                free(agent->orch_addr);
+                agent->orch_addr = NULL;
+        }
+        if (!ensure_orch_address(agent)) {
```

Comment:
I see there is also `ensure_orch_address` in `agent_start()` looks like we are duplicating the code? 

I also see a start duplicated:
```
# ./builddir/src/agent/hirte-agent -c /etc/hirte/agent.conf
2023-06-21 20:28:49,699+0000 INFO	: Translated 'control.medogz.com' to '10.88.3.87'
2023-06-21 20:28:49,705+0000 INFO	: Translated 'control.medogz.com' to '10.88.3.87'
2023-06-21 20:28:49,705+0000 INFO	: Connecting to manager on tcp:host=10.88.3.87,port=842
2023-06-21 20:28:49,706+0000 INFO	: Connected to manager as 'node1'
```

Reply:
How can I achieve this controller migration during the start of hirte?

Reply:
I'd say its re-using it ;) The intention was to keep the early evaluation if the address is valid - fail early, fail fast and avoid connecting to systemd etc. in case the host is invalid. Didn't think about the double-log, though, which is a bit confusing... I'll reshuffle this. Thanks!

Reply:
Update: Introduced a `agent_reconnect` which is always called after the initial start (`agent_connect` only is called then). The reconnect re-resolves the ip address (setting the orch_address) and simply calls `agent_connect` then.